include/electionguard/collections.h

```diff
@@ -0,0 +1,93 @@
+/// @file collections.h
+#ifndef __ELECTIONGUARD_CPP_COLLECTIONS_H_INCLUDED__
+#define __ELECTIONGUARD_CPP_COLLECTIONS_H_INCLUDED__
+
+#include "export.h"
+#include "status.h"
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief A singly-linked list data structure composed of strings for keys and strings for values.
+ */
+typedef struct eg_linked_list_s eg_linked_list_t;
```

Comment:
forward declare the pointer type in the header, but it's implementation is in the C file so consumers of this header don't know anything about the internals of the object structure. we deliberated over this pattern versus structure marshalling across the api surface and landed here because it was too difficult to maintain aligned data structures on both the C# side and the c/c++ side. this way is a bit more verbose on the interface specification but it allows the layout of the memory to change without impacting the public api surface. the world is full of compromises.